### PR TITLE
Add unit tests for Brave Ads country/subdivision codes

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/12828

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/tokens/redeem_unblinded_token/create_confirmation_util.cc [hunk 1]
```diff
@@ -84,11 +84,11 @@ std::string CreateConfirmationRequestDTO(
     const std::string locale =
         brave_l10n::LocaleHelper::GetInstance()->GetLocale();
 
-    if (locale::HasLargeAnonymity(locale)) {
+    if (locale::IsMemberOfAnonymitySet(locale)) {
       const std::string country_code = brave_l10n::GetCountryCode(locale);
       dto.SetKey("countryCode", base::Value(country_code));
     } else {
-      if (locale::IsAnonymous(locale)) {
+      if (locale::ShouldClassifyAsOther(locale)) {
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/tokens/redeem_unblinded_token/create_confirmation_util.cc:91]: @tmancey if a country is not part of the anonymity set nor classified as other we don't send a `country` key at all - is this still intended behaviour?

File: vendor/bat-native-ads/src/bat/ads/internal/locale/country_code_util.cc [hunk 2]
```diff
@@ -6,30 +6,30 @@
...
-#include "bat/ads/internal/locale/anonymous_country_codes.h"
-#include "bat/ads/internal/locale/large_anonymity_country_codes.h"
+#include "bat/ads/internal/locale/country_code_anonymity_set.h"
+#include "bat/ads/internal/locale/other_country_codes.h"
 
 namespace ads {
 namespace locale {
 
-bool HasLargeAnonymity(
+bool IsMemberOfAnonymitySet(
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/locale/country_code_util.cc:15]: good for now but thinking out loud: Right now the only variable by which we assign membership to the anonymity set is `country`. We'll likely be adding more soon, e.g. `channel` and `platform`. So `kCountryCodeAnonymitySet` and `kOtherCountryCodes` would give way to a more general approach.

File: vendor/bat-native-ads/src/bat/ads/internal/tokens/redeem_unblinded_token/create_confirmation_util.cc (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/tokens/redeem_unblinded_token/create_confirmation_util.cc:91]: This is still intended behaviour

**Review by moritzhaller — APPROVED** [review]
LGTM

**Comment by tmancey** [comment]
CI failed due to unrelated rewards browser tests

**Pull request merged**